src/pharmpy/modeling/basic_models.py

```diff
@@ -196,20 +292,31 @@ def _create_default_datainfo(path_or_df):
     return di
 
 
-def dosing(di: DataInfo, dataset, dose_comp: int):
-    # FIXME: Copied from plugins.nonmem.advan
-    if di is None:
-        return Bolus(sympy.Symbol('AMT'))
+def allowed_modeltype_administration(modeltype, administration):
```

Comment:
Why are we sometimes raising and sometimes returning from this function?

Reply:
Will change the function to be more uniform in how errors are raised

Reply:
However, if `administration` is the be removed, then this function can also be removed